app/jobs/webhook_job.rb

```diff
@@ -4,14 +4,21 @@ class WebhookJob < ApplicationJob
   def perform(payload, webhook_endpoint_id)
     webhook_endpoint = WebhookEndpoint.find(webhook_endpoint_id)
 
-    Typhoeus.post(
+    request = Typhoeus::Request.new(
       webhook_endpoint.target_url,
+      method: :post,
       headers: {
         "Content-Type" => "application/json; charset=utf-8",
         "X-Lapin-Signature" => OpenSSL::HMAC.hexdigest("SHA256", webhook_endpoint.secret, payload),
       },
       body: payload,
       timeout: TIMEOUT
     )
+
+    request.on_complete do |response|
+      throw "Webhook failed with status code #{response.code} - #{response.return_message} - #{response.body}" unless response.success?
```

Comment:
les rubyistes utilisent plutot `raise` que `throw`, les différences sont mineures - cf https://coderwall.com/p/lhkkug/don-t-confuse-ruby-s-throw-statement-with-raise

dans ce cas, la bonne pratique il me semble est de définir une exception custom : 
```
class OutgoingWebhookError < StandardError; end 
```

tout en haut de ce fichier, avant la definition de la classe du job. oui,oui, tu peux faire ça, tant qu'on n'a qu'une ou deux exceptions customs je pense que c'est OK, on pourra les regrouper un jour. 

tu peux ensuite faire

```
raise OutgoingWebhookError.new("#{response.code} - #{response.return_message} - #{response.body}")
```

et deux questions la dessus : 

- que renvoie `return_message` ? c'est une description textuelle du code HTTP ? si oui, ce n'est pas la peine je pense
- il faut tronquer le body, au cas ou ils nous repondent avec 2mo de texte. `response.body.truncate(length:100)` ou `response.body[0..100]` ou qqch comme ca

Si tu veux aller jusqu'à l'_extra mile_ et que ces metadonnées (surtout celle du code HTTP) apparaisse bien formattée dans Sentry, il faut definir une classe d'exception un peu plus spécialisée : https://blog.hashlabs.com/using-exception-context-in-sentry-with-rails-3dd5776c8b4e